src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,107 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final double LOAD_FACTOR = 0.75f;
+    private static final int DEFAULT_CAPACITY = 16;
+    private Node<K, V>[] table;
+    private int size;
+    private int threshold;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
+        if (size >= threshold) {
+            resize();
+        }
+        if (key == null) {
+            putForNullKey(value);
+            return;
+        }
+        int index = getIndex(key);
+        Node<K, V> currentNode = table[index];
 
+        while (currentNode != null) {
+            if (Objects.equals(key, currentNode.key)) {
+                currentNode.value = value;
+                return;
+            }
+            currentNode = currentNode.next;
+        }
+
+        addNode(key, value, index);
+        size++;
     }
 
     @Override
     public V getValue(K key) {
+        int index = getIndex(key);
+        Node<K, V> currentNode = table[index];
+
+        while (currentNode != null) {
+            if (Objects.equals(key, currentNode.key)) {
+                return currentNode.value;
+            }
+            currentNode = currentNode.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private void putForNullKey(V value) {
+        Node<K, V> currentNode = table[0];
+        while (currentNode != null) {
+            if (currentNode.key == null) {
+                currentNode.value = value;
+                return;
+            }
+            currentNode = currentNode.next;
+        }
+        addNode(null, value, 0);
+        size++;
+    }
+
+    private void resize() {
+        int newCapacity = table.length * 2;
+        threshold = (int) (newCapacity * LOAD_FACTOR);
+        Node<K, V>[] oldTable = table;
+        table = new Node[newCapacity];
+        size = 0;
+
+        for (Node<K, V> headNode : oldTable) {
+            while (headNode != null) {
+                put(headNode.key, headNode.value);
+                headNode = headNode.next;
+            }
+        }
+    }
+
+    private void addNode(K key, V value, int index) {
+        Node<K, V> newNode = new Node<>(key, value);
+        newNode.next = table[index];
+        table[index] = newNode;
+    }
+
+    private int getIndex(K key) {
+        return key == null ? 0 : Math.abs(key.hashCode()) % DEFAULT_CAPACITY;
```

Comment:
The getIndex method is using DEFAULT_CAPACITY instead of the current table length, which can lead to incorrect behavior after resize. Replace DEFAULT_CAPACITY with table.length to ensure correct index calculation.